q2.java

```diff
@@ -0,0 +1,113 @@
+
+
+//Implementing a linked list
+class LinkedList<T>{
+	Node<T> head;
+	
+	public LinkedList(){
+		head = null;
+	}
+	
+	public void appendLast(T data){
+		Node<T> last = new Node<T>(data);
+		Node<T> current = this.head;
+		if(current == null){
+			head = last;
+			return;
+		}
+		while(current.next!=null){

```

Comment:
Minor thing: try to be consistent when you write code. Here you didn't add any spaces either side of the !=, while above you did add spaces around the ==.
Personally I would write the line as 'while (current.next != null) {' with lots of spaces so each individual bit is easy to read. But as long as you are consistent across the while code that would be fine.